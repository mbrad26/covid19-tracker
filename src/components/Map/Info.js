import React from 'react';
import { Image } from 'react-bootstrap';

const Info = ({ activeCountry }) => {
  const styleDiv = {
    background: '#E9E8E6'
  }

  const styleParagraph = {
    fontSize: '.8rem',
    margin: '7px',
    fontWeight: 'bold'
  };

  return (
    <div style={styleDiv}>
      <h4>
        <Image
          src={activeCountry.countryInfo.flag}
          alt='flag'
        />
        <span> {activeCountry.country}</span>
      </h4>
      <hr />
      <p style={styleParagraph}>Cases: <span>{activeCountry.cases}</span></p>
      <p style={styleParagraph}>Critical: <span>{activeCountry.critical}</span></p>
      <p style={styleParagraph}>Deaths: <span>{activeCountry.deaths}</span></p>
      <p style={styleParagraph}>Deaths per 1 million: <span>{activeCountry.deathsPerOneMillion}</span></p>
      <p style={styleParagraph}>Recovered: <span>{activeCountry.recovered}</span></p>
      <p style={styleParagraph}>Nr of tests: <span>{activeCountry.tests}</span></p>
      <p style={styleParagraph}>Updated: <span>{new Date(activeCountry.updated).toDateString()}</span></p>
    </div>
  );
};

export default Info;
