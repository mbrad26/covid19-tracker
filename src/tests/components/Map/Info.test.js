import React from 'react';
import { render } from '@testing-library/react';

import { country } from '../../fixtures';
import Info from '../../../components/Map/Info';

describe('Info', () => {
  it('renders a snapshot', () => {
    const { container } = render(
      <Info activeCountry={country} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
