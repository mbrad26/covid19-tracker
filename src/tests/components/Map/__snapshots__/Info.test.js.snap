// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders a snapshot 1`] = `
<div
  style="padding: 0px 0px 0px 5px;"
>
  <h4>
    <img
      alt="flag"
      class=""
      src="https://disease.sh/assets/img/flags/al.png"
    />
    <span>
       
      Albania
    </span>
  </h4>
  <hr />
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Cases:          
    <span>
      9,513
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Critical:       
    <span>
      19
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Deaths:         
    <span>
      284
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Deaths/million: 
    <span>
      99
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Recovered:      
    <span>
      5,214
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Nr of tests:    
    <span>
      58,438
    </span>
  </p>
  <p
    style="font-size: .8rem; margin: 7px; font-weight: bold;"
  >
    Updated:        
    <span>
      Mon Aug 31 2020
    </span>
  </p>
</div>
`;
